galera/src/replicator_smm.cpp

```diff
@@ -259,6 +259,11 @@ galera::ReplicatorSMM::ReplicatorSMM(const struct wsrep_init_args* args)
     cert_.assign_initial_position(seqno, trx_proto_ver());
 
     build_stats_vars(wsrep_stats_);
+
+    // Now we have store directory name to conf. This directory name
+    // could later be used by other components, for example by gcomm
+    // to find appropriate location for view state file.
+    config_.set(BASE_DIR, data_dir_);
```

Comment:
It is Replicator object constructor, if it does not run completely we fail badly and its config object is destroyed.
But i agree if we have dedicated function to initialize configuration it should take care of base_dir initialization and once we have it in config_ we do not need to keep in the object itself.
Comment is addressed in commit https://github.com/codership/galera/commit/7f191981341e8629e7eaa557c50ae28f7494bafa